server/graphql/types/src/types/program/encounter.rs

```diff
@@ -101,25 +101,29 @@ pub struct EncounterFilterInput {
     pub clinician_id: Option<EqualFilterStringInput>,
     pub document_name: Option<EqualFilterStringInput>,
     pub document_data: Option<StringFilterInput>,
+    pub patient: Option<PatientFilterInput>,
+    pub program: Option<ProgramEnrolmentFilterInput>,
 }
 
-impl EncounterFilterInput {
-    pub fn to_domain_filter(self) -> EncounterFilter {
+impl From<EncounterFilterInput> for EncounterFilter {
+    fn from(f: EncounterFilterInput) -> Self {
         EncounterFilter {
-            id: self.id.map(EqualFilter::from),
-            patient_id: self.patient_id.map(EqualFilter::from),
-            program_id: self.program_id.map(EqualFilter::from),
-            created_datetime: self.created_datetime.map(DatetimeFilter::from),
-            start_datetime: self.start_datetime.map(DatetimeFilter::from),
-            status: self
+            id: f.id.map(EqualFilter::from),
+            patient_id: f.patient_id.map(EqualFilter::from),
+            program_id: f.program_id.map(EqualFilter::from),
+            created_datetime: f.created_datetime.map(DatetimeFilter::from),
+            start_datetime: f.start_datetime.map(DatetimeFilter::from),
+            status: f
                 .status
                 .map(|s| map_filter!(s, EncounterNodeStatus::to_domain)),
-            end_datetime: self.end_datetime.map(DatetimeFilter::from),
-            clinician_id: self.clinician_id.map(EqualFilter::from),
-            document_type: self.r#type.map(EqualFilter::from),
-            document_name: self.document_name.map(EqualFilter::from),
-            document_data: self.document_data.map(StringFilter::from),
+            end_datetime: f.end_datetime.map(DatetimeFilter::from),
+            clinician_id: f.clinician_id.map(EqualFilter::from),
+            document_type: f.r#type.map(EqualFilter::from),
+            document_name: f.document_name.map(EqualFilter::from),
+            document_data: f.document_data.map(StringFilter::from),
             program_context_id: None,
+            patient: f.patient.map(PatientFilterInput::into),
+            program: f.program.map(ProgramEnrolmentFilterInput::into),
```

Comment:
Think `program_enrolment` is a better name, e.g. it's named like this elsewhere...